File: 129. Sum Root to Leaf Numbers.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private int answer = 0;
    public int sumNumbers(TreeNode root) {
        helper(root, 0);
        return answer;
    }
    private void helper(TreeNode root, int current) {
        if (root == null) return;
        current *= 10;
        current += root.val;
        if (root.left == null && root.right == null) {
            answer += current;
            return;
        }
        helper(root.left, current);
        helper(root.right, current);
    }
}

File: 129. Sum Root to Leaf Numbers.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumNumbers(self, root: Optional[TreeNode]) -> int:
        answer = 0
        def dfs(root, sum):
            nonlocal answer
            if not root:
                return
            sum = sum*10 + root.val
            if not root.left and not root.right:
                answer += sum
                return
            dfs(root.left, sum)
            dfs(root.right, sum)
        dfs(root, 0)
        return answer